src/components/PostDetails.tsx

```
@@ -1,106 +1,128 @@
+  setIsFormOpen,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
 
-        <div className="block">
-          <Loader />
+  const [isLoading, setIsLoading] = useState(false);
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+  const [errorMessage, setErrorMessage] = useState(false);
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+  const loadComment = async () => {
+    setIsLoading(true);
 
-          <p className="title is-4">Comments:</p>
+    try {
+      const fetchedComment = await getPostComment(selectedPost.id);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+      setComments(fetchedComment);
+    } catch (error) {
+      setErrorMessage(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+  useEffect(() => {
+    loadComment();
+  }, [selectedPost]);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+  const handleDeleteComment = async (commentId: number) => {
+    try {
+      await deletePostComment(commentId);
+      setComments(prev => prev?.filter(c => c.id !== commentId) || []);
```

Issue: The fallback value for `setComments` after deleting a comment is currently `[]` if `prev` is falsy, but `prev` should always be an array. This fallback is unnecessary and could hide bugs. Please update to `setComments(prev => prev.filter(c => c.id !== commentId));` so that the state is always an array and the logic is clear.